corpus/data_augmentation.py

```diff
@@ -236,3 +243,167 @@ def run(self):
+            ir_path=self.ir_path, p=self.augment_prob
+        )
+
+        job_id = os.path.basename(self.job_id())
+        orig_corpus = corpus.Corpus()
+        new_corpus = corpus.Corpus()
+
+        orig_corpus.load(tk.get_path(self.corpus_file))
+        new_corpus.name = self.new_corpus_name
+        new_corpus.speakers = list(orig_corpus.all_speakers())
+        new_corpus.default_speaker = orig_corpus.default_speaker
+        new_corpus.speaker_name = orig_corpus.speaker_name
+
+        for r in orig_corpus.all_recordings():
+            perturbed_audio_name = "perturbed_" + r.audio.split("/")[-1]
+            (
+                samples,
+                sample_rate,
+            ) = audiomentations.core.audio_loading_utils.load_wav_file(
+                r.audio, sample_rate=None
+            )
+            augmented_samples = add_impulse_response(samples, sample_rate)
+            wavfile.write(
+                self.out_audio_folder.get_path() + "/" + perturbed_audio_name,
+                sample_rate,
+                augmented_samples,
+            )
+
+            pr = corpus.Recording()
+            pr.name = r.name
+            pr.segments = r.segments
+            pr.speaker_name = r.speaker_name
+            pr.speakers = r.speakers
+            pr.default_speaker = r.default_speaker
+            pr.audio = self.out_audio_folder.get_path() + "/" + perturbed_audio_name
+            new_corpus.add_recording(pr)
+
+        new_corpus.dump(self.out_corpus.get_path())
+
+
+class MixNoiseJob(Job):
+    """
+    Add noise to each recording in the corpus with a given probability to be affected.
+    Supports .wav files.
+
+    This job is dependent on the audiomentations library.
+    """
+
+    def __init__(
+        self,
+        corpus_file,
+        new_corpus_name,
+        noise_files_dir,
+        min_snr=3,
+        max_snr=30,
+        augment_prob=1,
+        time_rqmt=12,
+    ):
+        """
+
+        :param Path corpus_file: Bliss corpus with wav files
+        :param str new_corpus_name: name of the new corpus
+        :param str noise_files_dir: path to the directory containing noise files
+        :param float min_snr: minimum signal-to-noise ratio to be used
+        :param float max_snr: maximum signal-to-noise ratio to be used
+        :param float augment_prob: probability of each audio to be augmented by noise mixing
+        :param int time_rqmt: time requirement for the Sisypus job
+        """
+        self.corpus_file = corpus_file
+        self.noise_files_dir = noise_files_dir
+        self.new_corpus_name = new_corpus_name
+
+        self.out_audio_folder = self.output_path("out_audio/", directory=True)
+        self.out_corpus = self.output_path("noise_mixed.corpus.xml.gz")
+        self.out_segment_file = self.output_path("noise_mixed.segments")
+
+        self.min_snr = min_snr
+        self.max_snr = max_snr
+        self.augment_prob = augment_prob
```

Comment:
Sorting of the `__init__`: first set the inputs, then the outputs.

```suggestion
        self.corpus_file = corpus_file
        self.new_corpus_name = new_corpus_name
        self.noise_files_dir = noise_files_dir
        self.min_snr = min_snr
        self.max_snr = max_snr
        self.augment_prob = augment_prob

        self.out_audio_folder = self.output_path("out_audio/", directory=True)
        self.out_corpus = self.output_path("noise_mixed.corpus.xml.gz")
        self.out_segment_file = self.output_path("noise_mixed.segments")
```